Amazon.Ion.ObjectMapper/IonSerializer.cs

```diff
@@ -161,97 +163,79 @@ public void Serialize<T>(Stream stream, T item)
 
         public void Serialize<T>(IIonWriter writer, T item)
         {
+            currentDepth++;

```

Comment:
I could be wrong here since I'm not familiar with this code base, but my understanding of "stepping in" for depth refers to nested types such as structs and arrays. Does this belong here or would it better belong in a StepIn method of some sort like in the Ion library?

Reply:
Are you suggesting a `StepIn` and `StepOut` function just to handle the increment/decrement logic?